<launch>

    <arg name="filename" default="gazebo_navigation"/>
    <arg name="map_file" default="$(find routing_driver)/map/$(arg filename).yaml"/>

    <node name="grid_routing" pkg="routing_driver" type="grid_routing" output="screen">
        <param name="save_filename" value="$(find routing_driver)/waypoints/$(arg filename)_grid.yaml"/>
        <param name="length" value="4"/>
    </node>

    <node name="check_mapgrid_info" pkg="routing_driver" type="check_mapgrid_info" output="screen"/>

    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />    
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find routing_driver)/rviz/grid_routing.rviz" />

</launch>